### DISCUSSION TITLE
Zitadel started showing Welcome back screen and requires email before auth to an app

### DISCUSSION THREAD

**Discussion by pschiffe** [zitadel/zitadel#7978] [Q&A] [answered]
Hello,

since some recent release, Zitadel started to show Welcome back screen where I have to input my email before the auth continues to the app, even while my password and 2fa session is still valid. Is it possible to turn it off? It's kind of annoying as it's showing way too often and I have only 1 account I use to login to apps.

Thanks.

**Comment by hifabienne** [comment]
Hei @pschiffe 
Can you maybe share a little more information?
What is the process you are doing?
Maybe you can even share a screen recording so we can better understand it?

**Reply by pschiffe** [reply]
Hi @hifabienne. I have couple of web applications configured via OIDC. Since v2.51 or maybe v2.50, when I want to access one of the applications, I have to enter my email address on the welcome screen. Even when the session for password or 2fa is still valid, I have to enter just the email. The URL of this page is `/ui/login/login?authRequestID=2680...323` After entering the email, it continues to the app. It wasn't doing it before.

![Screenshot 2024-05-21 at 11-11-03 Welcome Back!](https://github.com/zitadel/zitadel/assets/3899107/d224a737-d078-42fd-9f7b-70dfc5dfd17b)

**Reply by pschiffe** [reply]
I'm also seeing this issue: https://github.com/zitadel/zitadel/issues/7957
I don't know if that may be related.

**Reply by hifabienne** [reply]
How do you redirect to ZITADEL?
Do you send a login hint? 
Which screen did you see before? The select account screen?

With only this screenshot I can't tell you much, this just tells me that you are on the ZITADEL login, but I don't understand the whole process. 

Your user is already authenticated in your application? And does have a valid token? Why do you redirect to ZITADEL in that case?

**Reply by pschiffe** [reply]
This is happening when I'm not authenticated in the application, but I'm authenticated in Zitadel. Before, browser was still redirected to the Zitadel in that case, but it automatically redirected back to the destination application without any input. Now I get the welcome screen and I have to input the email (and no password or 2fa) before it continues back to the app.

Before this screen was showed only when I had to enter 2fa code or password, otherwise no Zitadel screen was shown.

I have various applications configured, it's the same for all of them. Mostly OIDC apps, one SAML app as well. Even the Zitadel web ui itself is doing this. I go to Zitadel  domain, the welcome screen is shown, I enter just the email, Zitadel is working fine. After 10-15 minutes I go to the Zitadel domain again, in the same browser, the welcome screen is shown again.

**Reply by hifabienne** [reply]
@livio-a @peintnermax did we change something on the behaviour?

**Reply by livio-a** [reply]
@pschiffe could you check if there are any logs regarding `auth.user_sessions` of they're up to date?
and which version of cockroach do you use?

cc @adlerhurst might be the stale issue?

**Reply by pschiffe** [reply]
I'm having also this issue: https://github.com/zitadel/zitadel/issues/7957
I'm seeing errors `"msg":"unable to update state","projection":"auth.user_sessions"`
CockroachDB 23.2.5

**Comment by pschiffe** [comment, marked as answer]
This bug was fixed by PR #7982 and released in version [v2.53.0](https://github.com/zitadel/zitadel/releases/tag/v2.53.0)
